bpf/lib/conntrack.h

```diff
@@ -669,30 +670,26 @@ static __always_inline int ct_create6(void *map, struct ipv6_ct_tuple *tuple,
 	cilium_dbg3(ctx, DBG_CT_CREATED6, entry.rev_nat_index, ct_state->src_sec_id, 0);
 
 	entry.src_sec_id = ct_state->src_sec_id;
-	if (map_update_elem(map, tuple, &entry, 0) < 0)
+	if (map_update_elem(map_main, tuple, &entry, 0) < 0)
 		return DROP_CT_CREATE_FAILED;
 
-	/* Create an ICMPv6 entry to relate errors */
-	struct ipv6_ct_tuple icmp_tuple = {
-		.nexthdr = IPPROTO_ICMPV6,
-		.sport = 0,
-		.dport = 0,
-		.flags = tuple->flags | TUPLE_F_RELATED,
-	};
+	if (map_related != NULL) {
```

Comment:
Wouldn't it be better to check for `dir != CT_SERVICE` instead? That would remove a need for a caller to decide whether a related entry is required or not.

ICMP for `ClusterIP` type services should not work either, as kube-proxy rules match on L4 protocol, so ping to `ClusterIP` does not work.

Reply:
Passing in NULL from call-sites like those that pass CT_SERVICE as dir, but also others is more flexible. I originally had the same thought as you, but them decided against it since we also create CT_INGRESS/CT_EGRESS entries from NodePort side that don't need these.

Reply:
> since we also create CT_INGRESS/CT_EGRESS entries from NodePort side that don't need these.

Following this logic, shouldn't we avoid creating related entries from `bpf_lxc.c` for `CT_EGRESS` when dst is to a  `ClusterIP` svc?

Reply:
Although only relevant to older kernels, but yes we can do that too. I'll add it.